### PR TITLE
implement validator as a common module

### PR DESCRIPTION
**Pull request by icebob**
## :memo: Description

Convert the current `validator` broker options to other module configuration (transporter, cacher...etc)

### :dart: Relevant issues
#771 

### :gem: Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### :scroll: Example code
```js
//moleculer.config.js
module.exports = {
    nodeID: "node-100",
    validator: {
        type: "FastestValidator",
        options: {
            useNewCustomCheckerFunction: true,
            defaults: { ... },
            messages: { ... }
        }
    }
}
``` 

## :checkered_flag: Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] **I have added tests that prove my fix is effective or that my feature works**
- [x] **New and existing unit tests pass locally with my changes**
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -1187,14 +1187,17 @@ declare namespace Moleculer {
...
+	class BaseValidator {
 		constructor();
 		init(broker: ServiceBroker): void;
 		compile(schema: GenericObject): Function;
 		validate(params: GenericObject, schema: GenericObject): boolean;
 		middleware(): ((handler: ActionHandler, action: ActionSchema) => any);
+		convertSchemaToMoleculer(schema: any): GenericObject;
 	}
 
+	class Validator extends BaseValidator {} // deprecated
```
Comment (shawnmcknight) [inline index.d.ts:1199]: This actually appears to be exported as the `FastestValidator` class constructor and not the `BaseValidator` class constructor (see index.js).

Comment (icebob) [reply index.d.ts:1199]: Yeah, you are right. Thank, I'm fixing.

File: index.d.ts [hunk 2]
```diff
@@ -777,7 +777,7 @@ declare namespace Moleculer {
 
 		cacher?: boolean | Cacher | string | GenericObject;
 		serializer?: Serializer | string | GenericObject;
-		validator?: boolean | Validator;
+		validator?: boolean | BaseValidator;
```
[the remark this answers is not in the document]
Comment (icebob) [reply index.d.ts:780]: Great, thanks!

File: index.d.ts (hunk 1 above)
Comment (icebob) [reply index.d.ts:1199]: I can't change it because there is no definition for Fastest Validator. Definition of Fastest same as BaseValidator.

**Pull request merged**